tango_ros_common/tango_ros_common/src/main/java/eu/intermodalics/tango_ros_common/ParameterNode.java

```diff
@@ -64,6 +64,31 @@ public void onStart(ConnectedNode connectedNode) {
         syncLocalPreferencesWithParameterServer();
     }
 
+    // When using simulated time, direct or indirect calls to "connectedNode.getCurrentTime()" such
+    // as connectedNode.getLog().error(e) throw NullPointerExeption.
```

Comment:
If logging the only issue, you can overwrite the log function to do a simple `System.out.println` if the clock is not ready yet. This is a more elegant solution than busy waiting.